kmod/core/core.c

```diff
@@ -661,31 +699,26 @@ static int kpatch_write_relocations(struct kpatch_module *kpmod,
        unsigned long core = (unsigned long)kpmod->mod->module_core;
        unsigned long core_size = kpmod->mod->core_size;
 #endif
-	unsigned long src;
+	bool vmlinux = !strcmp(object->name, "vmlinux");
 
 	list_for_each_entry(dynrela, &object->dynrelas, list) {
-		if (!strcmp(object->name, "vmlinux")) {
-			ret = kpatch_verify_symbol_match(dynrela->name,
-							 dynrela->src);
-			if (ret)
-				return ret;
+		if (vmlinux || !dynrela->external) {
```

Comment:
If I'm not mistaken, I think we can simplify this even further to
```
if (dynrela->external)
   /* call kpatch_find_external_symbol */
else
   /* call kpatch_find_object_symbol */
```
When `dynrela->external` is set, we assume the symbol is either exported or in the patch module itself (kpmod). For all other cases, it should suffice to use kpatch_find_object_symbol() (for vmlinux, or the object being patched (object->name))